api/query/cache/monitor/monitor.go

```diff
@@ -175,13 +195,23 @@ func (m *indexMonitor) start() error {
 	return nil
 }
 
+func (m *indexMonitor) check() {
+	heapBytes := m.rt.GetHeapBytes()
+	if heapBytes > m.maxHeapBytes {
+		m.logger.Errorf("Out of memory: %d > %d", heapBytes, m.maxHeapBytes)
```

Comment:
This is not part of the PR, but I think we shouldn't use `Errorf` here as this is expected to happen eventually; and perhaps we should change the wording of the message to something like "%d memory allocated, exceeding the threshold %d" to avoid confusion with actual, fatal OOM.

Reply:
Sounds good. We can make this a `Warningf` and soften the wording.